### ISSUE TITLE
CKAN on Ubunto exceptions with Mono 3.10 installed

### ISSUE THREAD

**Issue by jwvanderbeck** [KSP-CKAN/CKAN-support#52]
I am trying to use CKAN on Ubuntu Linux 14.10.  I followed, to the letter, the instructions on this page:
https://github.com/KSP-CKAN/CKAN-support/wiki/Installing-CKAN-on-Ubuntu-14.04

To build Mono 3.10.  However when I try to run CKAN I still get an exception.

agathorn@john-linux:~/Downloads$ mono --version
Mono JIT compiler version 3.10.0 (tarball Mon Jan  5 01:22:31 PST 2015)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
	TLS:           __thread
	SIGSEGV:       altstack
	Notifications: epoll
	Architecture:  amd64
	Disabled:      none
	Misc:          softdebug 
	LLVM:          supported, not enabled.
	GC:            sgen
agathorn@john-linux:~/Downloads$ mono ckan.exe 
exception inside UnhandledException handler: Object reference not set to an instance of an object

[ERROR] FATAL UNHANDLED EXCEPTION: System.TypeInitializationException: An exception was thrown by the type initializer for System.Windows.Forms.XplatUI ---> System.TypeInitializationException: An exception was thrown by the type initializer for System.Drawing.GDIPlus ---> System.DllNotFoundException: libgdiplus.so
  at (wrapper managed-to-native) System.Drawing.GDIPlus:GdiplusStartup (ulong&,System.Drawing.GdiplusStartupInput&,System.Drawing.GdiplusStartupOutput&)
  at System.Drawing.GDIPlus..cctor () [0x00000] in <filename unknown>:0 
  --- End of inner exception stack trace ---
  at System.Drawing.Graphics.FromHdcInternal (IntPtr hdc) [0x00000] in <filename unknown>:0 
  at System.Windows.Forms.XplatUIX11.SetDisplay (IntPtr display_handle) [0x00000] in <filename unknown>:0 
  at System.Windows.Forms.XplatUIX11..ctor () [0x00000] in <filename unknown>:0 
  at System.Windows.Forms.XplatUIX11.GetInstance () [0x00000] in <filename unknown>:0 
  at System.Windows.Forms.XplatUI..cctor () [0x00000] in <filename unknown>:0 
  --- End of inner exception stack trace ---
  at System.Windows.Forms.Application.EnableVisualStyles () [0x00000] in <filename unknown>:0 
  at CKAN.GUI.Main (System.String[] args) [0x00000] in <filename unknown>:0 
  at CKAN.CmdLine.MainClass.Gui (System.String[] args) [0x00000] in <filename unknown>:0 
  at CKAN.CmdLine.MainClass.Main (System.String[] args) [0x00000] in <filename unknown>:0

**Comment by AlexanderDzhoganov** [comment]
Hmm, maybe try `apt-get install libgdiplus`?

**Comment by jwvanderbeck** [comment]
Ok so it turns out I was missing lots of dependencies.  Instead of following the instructions on the CKAN page for building and installing Mono 3.10, I instead went to the official Mono website, and followed the instructions there for adding the proper package.  When I did this and then installed the proper package, the package manager was able to see that I was missing a TON of dependencies and install those for me as well.  Now it all works.

tl;dr The CKAN page should **really** direct the user to proper install docs rather than a hacky way of just installing mono by itself.  This was my experience on a brand new fresh install of Ubuntu set up specific for KSP.

**Comment by hakan42** [comment]
You are welcome to provide a better text :smiling_imp:

**Comment by jwvanderbeck** [comment]
The better text is replace the entire webpage with a link to the official documentation for doing it :)

**Comment by AlexanderDzhoganov** [comment]
Sure, go ahead.
